torch/csrc/jit/passes/quantization/insert_quant_dequant.cpp

```diff
@@ -211,7 +248,10 @@ void insertQuantizationOps(
   }
   Value* original_val = observer->input(1);
   Node *quant, *choose_qparams, *dequant;
-  if (quant_type == QuantType::DYNAMIC && !isWeight(module, observer_out)) {
+  if (quant_type == QuantType::DYNAMIC && isNoopObserver(observer->input(0))) {
+    dequant = insertFP16CastOps(g, observer_out);
+  } else if (
+      quant_type == QuantType::DYNAMIC && !isWeight(module, observer_out)) {
     Value* dtype = g->insertGetAttr(self, qparam_names.back());
     std::tie(choose_qparams, quant, dequant) = insertChooseQParamQuantDequant(
         g, observer_out, dtype, at::Symbol::aten(quantize_func));
```

Comment:
probably better to add a comment for else branch, it's accepting both dynamic + activation and static quant